lib/ToolbarPanelHub.jsm

```diff
@@ -34,12 +34,25 @@ class _ToolbarPanelHub {
     this._getMessages = getMessages;
     if (this.whatsNewPanelEnabled) {
       this.enableAppmenuButton();
+      // Listen for pref changes that could turn off the feature
+      Services.prefs.addObserver(WHATSNEW_ENABLED_PREF, this);
     }
   }
 
   uninit() {
     EveryWindow.unregisterCallback(TOOLBAR_BUTTON_ID);
     EveryWindow.unregisterCallback(APPMENU_BUTTON_ID);
+    Services.prefs.removeObserver(WHATSNEW_ENABLED_PREF, this);
+  }
+
+  observe(aSubject, aTopic, aPrefName) {
+    switch (aPrefName) {
+      case WHATSNEW_ENABLED_PREF:
+        if (!this.whatsNewPanelEnabled) {
+          this.uninit();
```

Comment:
`uninit()` also removes the pref observer. So it would miss if the pref gets re-enabled after.

Reply:
Fixing this case is a little more tricky because `init` happens in `ASRouter` where the `getMessages` method is passed down. In the end I decided it's not worth covering this scenario because it doesn't seem like something we'll run into.
You can either:
* have the pref enabled -> it gets disabled remotely and whats new will cleanup
* have the pref disabled -> it gets enabled remotely 

Since this is not a user facing pref going through these states multiple times is not possible.